From: Ingo Saitz <ingo@debian.org>
Date: Sat, 29 Oct 2011 14:47:24 +0200
Subject: Use _LP64 to detect 64bit arches.

libmikmod-3.1.11a misses some checks for general 64bit support.
Upstream checks __x86_64, we do check _LP64, too. See bug #272457 and
http://www.cs.man.ac.uk/solaris/sun_docs/C/solaris_9/SUNWdev/SOL64TRANS/p16.html
---
 include/mikmod.h.in        |    2 +-
 include/mikmod_internals.h |    2 +-
 macintosh/mikmod.h         |    2 +-
 win32/mikmod_build.h       |    2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/mikmod.h.in b/include/mikmod.h.in
index 8adecff..6408b43 100644
--- a/include/mikmod.h.in
+++ b/include/mikmod.h.in
@@ -85,7 +85,7 @@ typedef char CHAR;
 
 @DOES_NOT_HAVE_SIGNED@
 
-#if defined(__arch64__) || defined(__alpha) || defined(__x86_64)
+#if defined(__arch64__) || defined(__alpha) || defined(__x86_64) || defined(_LP64)
 /* 64 bit architectures */
 
 typedef signed char     SBYTE;      /* 1 byte, signed */
diff --git a/include/mikmod_internals.h b/include/mikmod_internals.h
index 32da563..02097b6 100644
--- a/include/mikmod_internals.h
+++ b/include/mikmod_internals.h
@@ -50,7 +50,7 @@ extern "C" {
 /*========== More type definitions */
 
 /* SLONGLONG: 64bit, signed */
-#if defined (__arch64__) || defined(__alpha)
+#if defined(__arch64__) || defined(__alpha) || defined(__x86_64) || defined(_LP64)
 typedef long		SLONGLONG;
 #define NATIVE_64BIT_INT
 #elif defined(__WATCOMC__)
diff --git a/macintosh/mikmod.h b/macintosh/mikmod.h
index d00f53d..2930955 100644
--- a/macintosh/mikmod.h
+++ b/macintosh/mikmod.h
@@ -85,7 +85,7 @@ typedef char CHAR;
 
 
 
-#if defined(__arch64__) || defined(__alpha)
+#if defined(__arch64__) || defined(__alpha) || defined(__x86_64) || defined(_LP64)
 /* 64 bit architectures */
 
 typedef signed char     SBYTE;      /* 1 byte, signed */
diff --git a/win32/mikmod_build.h b/win32/mikmod_build.h
index 5e77ff3..33e6641 100644
--- a/win32/mikmod_build.h
+++ b/win32/mikmod_build.h
@@ -85,7 +85,7 @@ typedef char CHAR;
 
 
 
-#if defined(__arch64__) || defined(__alpha)
+#if defined(__arch64__) || defined(__alpha) || defined(__x86_64) || defined(_LP64)
 /* 64 bit architectures */
 
 typedef signed char     SBYTE;      /* 1 byte, signed */
